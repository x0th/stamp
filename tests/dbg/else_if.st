Cat = Object^;
Dog = Object^;
Dog meows = false;
Duck = Object^;
Duck meows = false;

if (Dog.meows)
{
	Dog.type
}
else if (Duck.meows)
{
	Duck.type
} 
else 
{
	Cat.type
}
