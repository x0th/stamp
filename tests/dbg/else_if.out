STDOUT:
----------------------
Context:
Callable = Callable-hash
Char = Char-hash
Int = Int-hash
List = List-hash
Object = Object-hash
Operators = Operators-hash
String = String-hash
false = false
if = if
true = true
----------------------
TokSList()
  TokSend()
    TokObject(Object)
    TokMessage(clone)
      TokObject(Cat)
  TokSend()
    TokObject(Object)
    TokMessage(clone)
      TokObject(Dog)
  TokStore()
    TokObject(Dog)
    TokValue(meows)
    TokObject(false)
  TokSend()
    TokObject(Object)
    TokMessage(clone)
      TokObject(Duck)
  TokStore()
    TokObject(Duck)
    TokValue(meows)
    TokObject(false)
  TokSend()
    TokObject(if)
    TokMessage(if_true)
      TokSList()
        TokSend()
          TokObject(Dog)
          TokMessage(meows)
        TokSList()
          TokSList()
            TokSend()
              TokObject(Dog)
              TokMessage(type)
        TokSend()
          TokObject(if)
          TokMessage(if_true)
            TokSList()
              TokSend()
                TokObject(Duck)
                TokMessage(meows)
              TokSList()
                TokSList()
                  TokSend()
                    TokObject(Duck)
                    TokMessage(type)
              TokSList()
                TokSend()
                  TokObject(Cat)
                  TokMessage(type)
Cat
----------------------
Context:
Callable = Callable-hash
Cat = Cat-hash
Char = Char-hash
Dog = Dog-hash
Duck = Duck-hash
Int = Int-hash
List = List-hash
Object = Object-hash
Operators = Operators-hash
String = String-hash
false = false
if = if
true = true
----------------------
STDERR:
